### DISCUSSION TITLE
Error in loading model

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#3171] [Q&A]
I want to load the old GCN model in my old code of GCN but this type of error occurs what can I do?

Error(s) in loading state_dict for GCNNet:
	Missing key(s) in state_dict: "Conv1.lin.weight", "Conv2.lin.weight", "Conv3.lin.weight". 
	Unexpected key(s) in state_dict: "Conv1.weight", "Conv2.weight", "Conv3.weight".

**Comment by wsad1** [comment]
This probably started happening after you updated to PyG2.0. In PyG2.0 the [GCNConvs](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/gcn_conv.html#GCNConv) parameters `self.weight` got changed to a linear layer `self.lin`.
I think changing the loaded parameter dictionary keys, should help.
```
params_dict = torch.load('model.pt')
# change keys and delete old keys. Do below for all relevant keys.
params_dict["Conv1.lin.weight"] = params_dict["Conv1.weight"]
del params_dict["Conv1.weight"]
modle.load_state_dict(params_dict) # if strict=Flase then old keys don't need to be deleted.
```

**Reply by shrimonmuke0202** [reply]
Also, the shape of the GCN layer is also mismatching if I load my old model in the new version of the PyG2.0 and the old model could not give the result that I want i.e. if I load the old model in the new PyG2.0. How can get rid of this type of ambiguity ?

**Reply by rusty1s** [reply]
You need to transpose the weights as well:
```
params_dict["Conv1.lin.weight"] = params_dict["Conv1.weight"].t()
```

**Reply by shrimonmuke0202** [reply]
But applying my save model on the old PyG 1.7.2 version gives the correct result but loading my model on the newer version does not give that result?

**Reply by shrimonmuke0202** [reply]
Now it is working, thanks for your answer
